### PR TITLE
encode timestamp with canonical format (iso8601) see #87

### PR DESCRIPTION
**Pull request by jlgeering**

### CODE REVIEW

**Comment by jlgeering** [comment]
should we also bump the version to 3.0?

**Review by mruoss — CHANGES_REQUESTED** [review]
The MR should at least add an entry to Changelog.md under the Unreleased section. Or we deal with versioning in this MR and create the 3.0.0 version... but mainly: please add a changelog entry.

File: lib/ymlr/encoder_test.exs [hunk 1]
```diff
@@ -199,7 +199,11 @@ defmodule Ymlr.EncoderTest do
     end
 
     test "datetime" do
-      assert MUT.to_s!(DateTime.from_naive!(~N[2016-05-24 13:26:08.003], "Etc/UTC")) == "2016-05-24 13:26:08.000000000 Z"
+      assert MUT.to_s!(~U[2016-05-24 13:26:08Z])      == "2016-05-24T13:26:08Z"
+      assert MUT.to_s!(~U[2016-05-24 13:26:08.1Z])    == "2016-05-24T13:26:08.1Z"
+      assert MUT.to_s!(~U[2016-05-24 13:26:08.02Z])   == "2016-05-24T13:26:08.02Z"
+      assert MUT.to_s!(~U[2016-05-24 13:26:08.003Z])  == "2016-05-24T13:26:08.003Z"
+      assert MUT.to_s!(~U[2016-05-24 13:26:08.0004Z]) == "2016-05-24T13:26:08.0004Z"
```
Comment (mruoss) [inline lib/ymlr/encoder_test.exs:206]: Wanna add a non-UTC test case?

Comment (jlgeering) [reply lib/ymlr/encoder_test.exs:206]: DateTime only supports UTC, we would have to add a dependency to `tz` or `tzdata`, see https://hexdocs.pm/elixir/1.13/DateTime.html#module-time-zone-database

(i.e. I try to do exactly that and the compiler would not let me)

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Changed
 
 - use `Enum.map_join/3` indead of `Enum.map/2` and `Enum.join/2` as it's more efficient according to credo recommendations
+- fix #87 - change the serialization of timestamps to use the canonical (iso8601) format, i.e. before: `2022-07-31 14:48:48.000000000 Z` and now: `"2022-07-31T14:48:48Z"`
```
Comment (mruoss) [inline CHANGELOG.md:12]: While this (#87) might work inside Github, would be nice to make this a link also on hexdocs, no? i.e. `[normal markdown link](url)`

Comment (mruoss) [reply CHANGELOG.md:12]: Otherwise good to go and then let's bundle 3.0.0?

**Review by mruoss — COMMENTED** [review]
Otherwise good to go and then let's bundle 3.0.0?

**Review by mruoss — DISMISSED** [review]

**Review by mruoss — CHANGES_REQUESTED** [review]
Some more context: support for DateTime was added in #17 by @joerichsen.

Also, now that we don't use the Calendar module anymore, this version is gonna work on Elixir 1.10 again. Can you add it back to the GH action Matrix and update the README, @jlgeering ?

**Review by mruoss — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -11,9 +11,17 @@ jobs:
...
-        elixir: [1.11.x, 1.12.x]
-        otp: [22.x, 23.x, 24.x]
+        elixir: ['1.11.x', '1.12.x', '1.13.x']
+        otp: ['22.x', '23.x', '24.x']
+        include:
+          - elixir: '1.10.x'
+            otp: '22.x'
+          - elixir: '>1.10.3'
+            otp: '23.x'
+          - elixir: '1.13.4'
```
Comment (mruoss) [inline .github/workflows/test.yaml:23]: can be 1.13.x, no? I would only run on the latest patch versions... and otp 25 is supported _starting_ from 1.13.4, but all later patches are going to support otp 25.

**Pull request merged**